src/main/webapp/app/exercises/programming/manage/update/update-components/information/programming-exercise-information.component.ts

```diff
@@ -0,0 +1,219 @@
+import { AfterViewInit, Component, Input, OnChanges, OnDestroy, QueryList, SimpleChanges, ViewChild, ViewChildren, effect, inject, input, signal, viewChild } from '@angular/core';
+import { NgModel } from '@angular/forms';
+import { ProgrammingExercise, ProjectType } from 'app/entities/programming/programming-exercise.model';
+import { ProgrammingExerciseCreationConfig } from 'app/exercises/programming/manage/update/programming-exercise-creation-config';
+import { ExerciseTitleChannelNameComponent } from 'app/exercises/shared/exercise-title-channel-name/exercise-title-channel-name.component';
+import { Subject, Subscription } from 'rxjs';
+import { TableEditableFieldComponent } from 'app/shared/table/table-editable-field.component';
+import { every } from 'lodash-es';
+import { ImportOptions } from 'app/types/programming-exercises';
+import { ProgrammingExerciseInputField } from 'app/exercises/programming/manage/update/programming-exercise-update.helper';
+import { removeSpecialCharacters } from 'app/shared/util/utils';
+import { CourseExistingExerciseDetailsType, ExerciseService } from 'app/exercises/shared/exercise/exercise.service';
+
+const MAXIMUM_TRIES_TO_GENERATE_UNIQUE_SHORT_NAME = 200;
+
+@Component({
```

Comment:
Can we make this standalone as well?

Reply:
This leads to roughly 20 errors that would need to be fixed, so this is out of scope for this PR in my opinion as it has already reached a size that is rather big anyways.